src/main/java/org/springframework/data/couchbase/core/convert/CouchbaseConverter.java

```diff
@@ -54,4 +56,9 @@
 	 * @return the name of the field that will hold type information.
 	 */
 	String getTypeKey();
+
+	/**
+	 * @return the alias value for the type
+	 */
+	Alias getAlias(TypeInformation<?> info);
```

Comment:
fixed 
(the reason I used getAlias instead of getTypeAlias is because the implementation relies on a method org.springframework.data.convert.DefaultTypeMapper.getAliasFor(type) (not getTypeAliasFor) )   Although in that method it is clear from the argument that it is an alias for the type.

Reply:
so the method below is addressed, but not this one?

Reply:
other than that, looks good to me

Reply:
I refactored the one in CouchbaseTypeMapper and thought that was it. 
I've changed the definition here in CouchbaseConverter, the implementation in MappingCouchbaseConverter and use in Query.